### PR TITLE
feat: allow iteratees of orderBy to be functions

### PR DESCRIPTION
**Pull request by dated**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -13,8 +13,8 @@ export function sortByDesc<T = any>(values: any, iteratees?: any): T[] {
     return sort(values).desc(iteratees);
 }
 
-export function orderBy<T = any>(values: T[], iteratees: string | string[], orders: string | string[]): T[] {
-    if (typeof iteratees === "string") {
+export function orderBy<T = any>(values: T[], iteratees: Function | string | any[], orders: string | string[]): T[] {
```
Comment (faustbrian) [inline src/index.ts]:
```suggestion
export function orderBy<T = any>(values: T[], iteratees: Function | string | string[], orders: string | string[]): T[] {
```

Comment (dated) [reply src/index.ts]: this wouldn't allow passing in multiple functions, see the last commit.

File: src/index.ts [hunk 2]
```diff
@@ -13,9 +15,15 @@ export function sortByDesc<T = any>(values: any, iteratees?: any): T[] {
     return sort(values).desc(iteratees);
 }
 
-export function orderBy<T = any>(values: T[], iteratees: string | string[], orders: string | string[]): T[] {
+export function orderBy<T = any>(
+    values: T[],
+    iteratees: CallbackFunctionVariadicAnyReturn | string | CallbackFunctionVariadicAnyReturn[] | string[],
```
Comment (faustbrian) [inline src/index.ts]: What would be an example of `CallbackFunctionVariadicAnyReturn[]` as use-case for sorting?

Comment (dated) [reply src/index.ts]: On second thought this is probably unneeded, since all sorting logic can be passed in one function. I'll remove that asap.

**Pull request merged**